doc/modules/linear_model.rst

```diff
@@ -842,6 +848,98 @@ Ridge Regression`_, see the example below.
 
     .. [4] Tristan Fletcher: `Relevance Vector Machines Explained <https://citeseerx.ist.psu.edu/doc_view/pid/3dc9d625404fdfef6eaccc3babddefe4c176abd4>`_
 
+.. _linear_regression_intercept:
+
+Note on the handling of the intercept in linear regression models
+=================================================================
+
+All least-squares linear regression estimators in scikit-learn solve a centered
+version of the problem without intercept and then compute the intercept a
+posteriori from that solution.
+
+This ensures that the ridge solution converges to the minimal norm OLS solution
+when `alpha` tends to zero in the underdeterimined case (more features than
+samples).
```

Comment:
Could you elaborate on this? Do we have a problem when we include the intercept in the norm? Does Ridge then no more converge to the same solution?

Reply:
I believe so because we do not penalize the intercept parameter in the ridge objective and the ridge solutions should be a continuous function of alpha. But I do that have a formal proof beyond this intuition. I plan to do some experiments and maybe a new test.

Reply:
It seems like your central argument to exclude the intercept from that norm. Would be nice to support this hypothesis.

From a testing point of view, it is easier to add a column of ones to X and check for an exact match of coefficients than doing a limit of Ridge with tiny alpha.

Reply:
I agree. I thought it was intuitive but it's not in retrospect. I'll do that ASAP.

Reply:
I did the experiments. Let's move this discussion to the main comment area of the PR for the sake of visibility:

https://github.com/scikit-learn/scikit-learn/pull/25948#issuecomment-1517587995